language/move-compiler/src/naming/fake_natives.rs

```diff
@@ -86,37 +87,26 @@ pub fn resolve_builtin(
     function: &FunctionName,
 ) -> Option<fn(Vec<IR::Type>) -> IR::Bytecode_> {
     let sp!(_, ModuleIdent_ { address, module }) = module;
-    let addr_name = match address {
-        Address::Numerical(Some(sp!(_, n_)), _) | Address::NamedUnassigned(sp!(_, n_)) => n_,
-        _ => return None,
+    // Only resolve if either (a) the address is named "std" or (b) its value is 0x1
+    match address {
+        Address::Numerical(Some(sp!(_, n)), _) | Address::NamedUnassigned(sp!(_, n))
+            if n.as_str() == "std" => {}
+        Address::Numerical(_, sp!(_, a)) if a.into_inner() == AccountAddress::ONE => {}
+        _ => {
+            return None;
+        }
```

Comment:
After doing some more digging, some thoughts here:

In my pervious comment, I was wondering if we could check if `std` == `0x1`. Unfortunately that isn't really easy after the address-map became a per-package binding. So we could thread that through to here but that is way more work than we we need right now. 

In an early version of this design, I considered doing something like `#[bytecode_instruction(VecBorrow)]` instead of just `#[bytecode_instruction]` and then not even checking things like the function name, but maybe doing some sanity checking for signature. That seemed like overkill at the time. But it would help here for this case as we wouldn't need these extra checks.
What I'm getting at is the motivation of all of these checks is to prevent unexpected behavior from someone programming against a module. We have the `native` check for that. We have the `#[bytecode_instruction]` for that. 

I don't really care at this point if we also have the address check for that. 

So we can either:
1) Land this code as is, and revisit it if anyone ever wants to put `std` at something other than 0x1
2) Just delete this check

Reply:
I guess let's go with (1) is the safest. I'm not sure about the consequences removing this check, with all this fuzz lately, keeping the check seems better. I believe a user _can_ actually locate `std` elsewhere than `0x1` with this code.